SU2_CFD/include/numerics/CFileReaderLUT.hpp

```diff
@@ -0,0 +1,103 @@
+/*!
+ * \file CFileReaderLUT.hpp
+ * \brief reading lookup table for tabulated fluid properties
+ * \author D. Mayer, T. Economon
+ * \version 7.3.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "../../Common/include/parallelization/mpi_structure.hpp"
+
+using namespace std;
+
+class CFileReaderLUT {
+ protected:
+  int rank;
+
+  string type_lut;
+  string version_lut;
+  string version_reader;
+  unsigned long n_points;
+  unsigned long n_triangles;
+  unsigned long n_hull_points;
+  unsigned long n_variables;
+
+  /* !brief
+   * Holds the variable names stored in the table file. Order is in sync with
+   * tableFlamelet.
+   */
+  vector<string> names_var;
+
+  /* !brief
+   * Holds all data stored in the table. First index addresses the variable
+   * while second index addresses the point.
+   */
+  vector<vector<su2double> > table_data;
+
+  vector<vector<unsigned long> > triangles;
```

Comment:
Use su2matrix, vector of vectors is a bad format

Reply:
@pcarruscag what would be the equivalent of these:
table_data.at(ixEnth)
table_data.at(ixEnth).begin()

I thought it would be table_data(ixEnth) but I guess I don't understand what's going on in C2DContainer